client/test/router_store.spec.js

```diff
@@ -0,0 +1,669 @@
+      'mailbox/:mailboxID/:messageID/forward': 'messageForward',
+      'mailbox/:mailboxID/:messageID/reply': 'messageReply',
+      'mailbox/:mailboxID/:messageID/reply-all': 'messageReplyAll',
+      'mailbox/:mailboxID/:conversationID/:messageID(?:query)': 'messageShow',
+      '': 'defaultView',
+    },
+  },
+};
+
+
+describe('Router Store', () => {
+  let routerStore;
+  let dispatcher;
+
+
+  function changeRoute(query, message) {
+    if (message === undefined) message = fixtures.message1;
+    dispatcher.dispatch({
+      type: ActionTypes.ROUTE_CHANGE,
+      value: {
+        accountID: fixtures.account.id,
+        mailboxID: fixtures.account.inboxMailbox,
+        tab: 'selected',
+        action: MessageActions.SHOW,
+        conversationID: message.conversationID,
+        messageID: message.id,
+        query,
+      },
+    });
+  }
+
+  function loadMessages(messages, conversationLength) {
+    dispatcher.dispatch({
+      type: ActionTypes.MESSAGE_FETCH_SUCCESS,
+      value: {
+        lastPage: fixtures.lastPage,
+        result: {
+          messages,
+          conversationLength,
+        },
+        timestamp: new Date(),
+      },
+    });
+  }
+
+  function generateMessages() {
+    const msgs = [];
+    for (let i = 0; i < UtilConstants.MSGBYPAGE + 3; i++) {
+      const message = _.clone(fixtures.message1);
+      message.id = `id${i}`;
+      message.messageID = `meid${i}`;
+      message.conversationID = `c${i}`;
+      msgs.push(message);
+    }
+    return msgs;
+  }
+
+  /*
+   * Problem noticed:
+   *
+   * * FIXME: Private methods are melted with public ones.
+   * * FIXME: Store requests other stores without using getters.
+   * * FIXME: Some events require a value while it's never used
+   *          (ex: EDIT_ACCOUNT_REQUEST).
+   * * FIXME: EDIT_ACCOUNT_SUCCESS clears errors after creating some. Don't
+   *          know what to test.
+   * * FIXME: Code is shared bewteen client and server.
+   * * FIXME: getPreviousConversation and getNextConversation looks confusing.
+   */
+
+  before(() => {
+    dispatcher = new SpecDispatcher();
+    mockeryUtils.initDispatcher(dispatcher);
+    mockeryUtils.initForStores([
+      '../app/stores/router_store',
+      '../stores/account_store',
+      '../stores/message_store',
+      '../../../server/utils/constants',
+    ]);
+    routerStore = require('../app/stores/router_store');
+  });
+
+  describe('Actions', () => {
+    it('ROUTE_INITIALIZE', () => {
+      dispatcher.dispatch({
+        type: ActionTypes.ADD_ACCOUNT_SUCCESS,
+        value: { account: _.clone(fixtures.account) },
+      });
+      dispatcher.dispatch({
+        type: ActionTypes.RECEIVE_RAW_MESSAGES,
+        value: [fixtures.message1, fixtures.message2],
+      });
+      dispatcher.dispatch({
+        type: ActionTypes.ROUTES_INITIALIZE,
+        value: fixtures.router,
+      });
+      assert.equal(routerStore.getRouter(), fixtures.router);
```

Comment:
I don't understand the link between this line and the 3 dispatches above. Can you explain it to me?